packages/easysearch:core/lib/engines/external-index.js

```diff
@@ -81,10 +60,45 @@ class ExternalEngine extends ReactiveEngine {
       sort: this.callConfigMethod('sort', searchDefinition, options),
       limit: options.search.limit,
       skip: options.search.skip,
-      fields: this.callConfigMethod('fields', searchDefinition, options)
+      fields: options.index.fields
     };
   }
 
+  /**
+   * Return array with fetched data
+   *
+   * @param {Object} selector Filter for api
+   * @param {String} searchString Search definition
+   * @param {Object} findOptions          Search and index options
+   */
+  externalFetch(selector, searchString, findOptions) {
+  /*
+      filter: {javaClass: 'java.util.HashMap', map: {
+          'details.userId': this.userId,
+      }}
+  */
+    console.log('findOptions', findOptions);
+    const args = [searchString, findOptions.fields, JSON.stringify(selector), findOptions.skip, findOptions.limit];
+    console.log('args', args);
+    return Promise.await(ExGuardianApi.call('elasticSearch.mongoCustomSearch', args));
```

Comment:
We need to cancel pending promises if we initiate new search, otherwise we trigger too many requests to the backend and then receive data that we are not interested anymore. Check 'bluebird' promise cancel method